tests/unit/test_application.py

```diff
@@ -1121,29 +1350,88 @@ def grammar_project(tmp_path):
     project_file.write_text(contents)
 
 
+@pytest.fixture()
+def non_grammar_project_full(tmp_path):
+    """A project that builds on amd64 to riscv64."""
+    project_file = tmp_path / "testcraft.yaml"
+    project_file.write_text(FULL_PROJECT_YAML)
+
+
+@pytest.fixture()
+def grammar_project_full(tmp_path):
+    """A project that builds on amd64 to riscv64 and s390x."""
+    project_file = tmp_path / "testcraft.yaml"
+    project_file.write_text(FULL_GRAMMAR_PROJECT_YAML)
+
+
+@pytest.fixture()
+def non_grammar_build_plan(mocker):
+    """A build plan to build on amd64 to riscv64."""
+    host_arch = "amd64"
+    base = util.get_host_base()
+    build_plan = [
+        models.BuildInfo(
+            f"platform-{build_for}",
+            host_arch,
+            build_for,
+            base,
+        )
+        for build_for in ("riscv64",)
+    ]
+
+    mocker.patch.object(MyBuildPlanner, "get_build_plan", return_value=build_plan)
+
+
 @pytest.fixture()
 def grammar_build_plan(mocker):
     """A build plan to build on amd64 to riscv64 and s390x."""
     host_arch = "amd64"
     base = util.get_host_base()
-    build_plan = []
-    for build_for in ("riscv64", "s390x"):
-        build_plan.append(
-            models.BuildInfo(
-                f"platform-{build_for}",
-                host_arch,
-                build_for,
-                base,
-            )
+    build_plan = [
+        models.BuildInfo(
+            f"platform-{build_for}",
+            host_arch,
+            build_for,
+            base,
         )
+        for build_for in ("riscv64", "s390x")
+    ]
 
     mocker.patch.object(MyBuildPlanner, "get_build_plan", return_value=build_plan)
 
 
 @pytest.fixture()
-def grammar_app(
+def grammar_app_mini(
+    tmp_path,
+    grammar_project_mini,  # noqa: ARG001
+    grammar_build_plan,  # noqa: ARG001
```

Comment:
```suggestion
@pytest.mark.usefixtures("grammar_project_mini", "grammar_build_plan")
```

Reply:
While I got this: `pytest.mark.usefixtures` has no effect on fixtures

Reply:
You're right, I forgot that doesn't work for fixtures.